.github/workflows/release-check.yaml

```diff
@@ -0,0 +1,28 @@
+name: release-tests
+on: [push, pull_request]
+jobs:
+  check-tarball:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Cache cargo registry
+        uses: actions/cache@v4
+        with:
+          path: ~/.cargo/registry
+          key: ${{ runner.os }}-cargo-registry-${{ hashFiles('**/Cargo.lock') }}
+
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: "stable"
+          override: true
+
+      - name: make a release tarball and build from it
+        run: |
+          cmake -S. -Bbuild &&
+          make -Cbuild package_source &&
+          tar -xf build/task-*.tar.gz &&
+          cd task-*.*.* &&
+          cmake -S. -Bbuild &&
+          make -Cbuild task_executable
```

Comment:
Why are we using make rather than cmake here?

Reply:
Huh,
```
cmake -S. -Bbuild --target task_executable
```
doesn't work, but omitting `-S.` fixes it:
```
cmake -Bbuild --target task_executable
```
so I guess we can use that!

Reply:
Oh, `-B` and `--build` aren't the same!